os-eap-migration/added/launch/openshift-migrate-common.sh

```diff
@@ -29,8 +30,15 @@ function runMigration() {
   local success=false
   local message="Finished, migration pod has been terminated"
   ${JBOSS_HOME}/bin/readinessProbe.sh
+  local probeStatus=$?
 
-  if [ $? -eq 0 ] ; then
+  if [ $probeStatus -eq 0 ] && [ "$(type -t probePodLog)" = 'function' ]; then
```

Comment:
Does this check need to be able to discern the difference between an application failure and a probe failure (e.g. probe fails because server's taking too long to start)?

Reply:
agree, the check currently is not capable to discern the probe and application failure. Is your point the application failure (aka ERROR during app startup) should not cause migrate pod to be marked as not capable to start, right?

Reply:
More that the probe might fail because the management port isn't available yet, e.g. if it were run immediately after startup, i.e. it's still initializing.  There is logic in the probe script to accommodate retries, e.g. it retries 30 times with a one second delay after each failure, but we're thinking of moving that configuration into the container's probe configuration directly, which could affect usage here.  This is just something to consider.

Reply:
I was (at least aware) of the fact there are some retries for the startup initialization. The function which pobes the log (aka. downloading the log by OpenShift API call) waits while the readiness probe returns success. At that point the script expects the container is started and it checks the content of the log.

I'm not sure what is meant by the "container's probe configuration". I expect such refactoring could affect this PR but it could be that not much. The script here just waits for get a started app server. If it's possible to find the status of the server (started ok/error on starting) then the functionality here will be just the same as it's presented in this PR.

Reply:
This is what I was referring to: https://docs.openshift.com/container-platform/3.9/dev_guide/application_health.html

In addition to what's listed there, you can also specify `periodSeconds`, `successThreshold` and `failureThreshold`, the first and last of which could replace the settings in the probe itself.  That's what I was talking about.

Reply:
I don't think anything needs to change.  I'm just pointing out that if we go this route, we need to ensure we preserve this type of functionality in the probes.  There are other products that are looking at switching to configuring some of this on the pods, because it works better for their use case.  That said, I think you could hard code those environment variables in your migration script to future proof yourself, should the defaults ever change (e.g. go to 1/1).

Reply:
I see. I'm just now thinking if it influences this verification of the server log content. I mean I named it as probe but it was not meant to be used for checking container readiness. It was meant to check if the migration action could be processed. The migration pod itself is already marked as live and ready but this part of checking plays role during the migration pod starting app server which then process the migration action. If the app container is not ready to run or there is some problem in server log it does not mean the migration pod is marked as "a not ready". It just means it tries to start the app server once again in a while.